runtime/compiler/optimizer/JProfilingRecompLoopTest.cpp

```diff
@@ -0,0 +1,245 @@
+ *                                  |                                                              |
+ *                                  V                                                              |
+ *             -----------------------------------------------                                     |
+ *             |ifcmple goto remainingCodeBlock              | <---- CalculateCheckLoopFreqBlock   |
+ *             |   rawFrequencyOfTheLoop                     |                                     |
+ *             |   AdjustedLoopRecompilationThreshold        |                                     |
+ *             -----------------------------------------------                                     |
+ *                                  |                                                              |
+ *                                  |                                                              |
+ *                                  V                                                              |
+ *             -----------------------------------------------                                     |
+ *             |call TR_jitRetranslateCallerWithPrep         | <--- callRecompilation block        |
+ *             |       loadaddr startPC                      |                                     |
+ *             |       loadaddr                              |                                     |
+ *             -----------------------------------------------                                     |
+ *                                  |                                                              |
+ *                                  |                                                              |
+ *                                  V                                                              |
+ *                   --------------------------------                                              |
+ *                   |treeTops after insertion Point|  <--- remainingCodeBlock                     |
+ *                   |from original block           |                                              |
+ *                   |...                           |<----------------------------------------------   
+ *                   --------------------------------
+ *    \param comp Current compilation object
+ *    \param testLocations RecompilationTestLocation list containing TreeTops after which test are required to be added
+ *                         And corresponding to that location, a loop nesting depth 
+ */
+
+void 
+TR_JProfilingRecompLoopTest::addRecompilationTests(TR::Compilation *comp, RecompilationTestLocationsInfo &testLocations)
+   {
+   TR_PersistentProfileInfo *profileInfo = comp->getRecompilationInfo()->findOrCreateProfileInfo();
+   TR_BlockFrequencyInfo *bfi = TR_BlockFrequencyInfo::get(profileInfo);
+   TR::CFG *cfg = comp->getFlowGraph();
+   cfg->invalidateStructure();
```

Comment:
Can we have a TODO to do a compile time evaluation to see if it is better to fix up structure or invalidate structure since fix-up should be possible given the nature of what is going on here.

Reply:
Added TODO